serverless.yml

```diff
@@ -0,0 +1,86 @@
+service:
+  name: superformula-cloud-backend-test
+
+frameworkVersion: ">=1.1.0 <=2.40.0"
+
+plugins:
+  - serverless-dynamodb-local
+  - serverless-dotenv-plugin
+  - serverless-offline
+  - serverless-plugin-typescript
+
+custom:
+  dynamodb:
+    stages:
+      - dev
+    start:
+      port: 8000
+      inMemory: true
+      migrate: true
+    # Comment if you don't have a DynamoDB running locally
+      noStart: true
+    migration:
+      dir: offline/migrations
+  user_dynamodb_table_name: user
+  othermodel_dynamodb_table_name: othermodel
+
+provider:
+  name: aws
+  runtime: nodejs12.x
+  timeout: 15
+  apiKeys:
+    - name: lambdaApolloKey
+      value: ${env:APOLLO_LAMBDA_KEY}
+  environment:
+    MAPBOX_ACCESS_TOKEN: "YOUR_KEY_HERE" #Public token
+    USER_DYNAMODB_TABLE: ${self:service.name}-${self:custom.user_dynamodb_table_name}-${opt:stage, self:provider.stage}
+    OTHERMODEL_DYNAMODB_TABLE: ${self:service.name}-${self:custom.othermodel_dynamodb_table_name}-${opt:stage, self:provider.stage}
+    APOLLO_KEY: "YOUR_KEY_HERE"
+    APOLLO_GRAPH_VARIANT: "staging"
+    APOLLO_SCHEMA_REPORTING: true
+  iamRoleStatements:
+    - Effect: Allow
+      Action:
+        - dynamodb:Query
+        - dynamodb:Scan
+        - dynamodb:GetItem
+        - dynamodb:PutItem
+        - dynamodb:UpdateItem
+        - dynamodb:DeleteItem
+      Resource: "arn:aws:dynamodb:${opt:region, self:provider.region}:*:table/${self:provider.environment.USER_DYNAMODB_TABLE}"
+
+functions:
+  graphql:
+    handler: src/handler.graphqlHandler
+    events:
+      - http:
+          private: true
+          path: graphql
+          method: post
+          # cors: true
```

Comment:
When would you want to toggle this on?

Reply:
If I decide that this service should allows public access. Since this service will be online just for the time of this hiring process takes and with the intention of simplifying it, I commented out the attribute.

Ususally if I would need to turn it on/of based on the environment (dev, staging, prod, etc) I would set and environment variable and decide the valie if this mechanism based on where the service is deployed.

Reply:
What would be the advantages and disadvantages of turning CORS off in dev/staging, and what would be your decision criteria for this configuration?